scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -101,6 +112,31 @@ static __noinline u32 iter_layer_cpu_ctx(u32 layer_idx, int idx)
 	return offset;
 }
 
+/*
+ * Returns the iterator context for iterating over DSQs using the configured
+ * DSQ iterator algorithm.
+ */
+static u32 iter_layer_dsq_ctx(int idx, u32 layer_idx)
+{
+	switch (dsq_iter_algo) {
+	case DSQ_ITER_LINEAR: {
+		return idx;
+	}
+	case DSQ_ITER_ROUND_ROBIN: {
+		return dsq_iter_rr_cpu_ctx(layer_idx, idx);
+	}
+	case DSQ_ITER_WEIGHT: {
+		return dsq_iter_weight_ctx(idx);
+	}
+	case DSQ_ITER_REVERSE_WEIGHT: {
+		return dsq_iter_weight_ctx(idx);
```

Comment:
This is unimplemented, I plan on making this a help wanted BPF task as it should be pretty straight forward.